lib/db_utils.go

```diff
@@ -8785,25 +8877,98 @@ func DBGetPostAssociationIdsByAttributes(handle *badger.DB, snap *Snapshot, asso
 			associationQuery.PostHash == nil ||
 			associationQuery.AssociationType == "" ||
 			associationQuery.AssociationValue == "" {
-			return nil, errors.New("DBGetPostAssociationIdsByAttributes: invalid query params")
+			return nil, nil, errors.New("DBGetPostAssociationIdsByAttributes: invalid query params")
 		}
 		keyPrefix = append(keyPrefix, associationQuery.AppPKID.ToBytes()...)
 	}
 
+	// Convert LastSeenAssociationID to LastSeenKey.
+	var lastSeenKey []byte
+	if associationQuery.LastSeenAssociationID != nil {
+		lastSeenKey, err = _dbPostAssociationIdToKey(handle, snap, associationQuery.LastSeenAssociationID, prefixType)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "DBGetPostAssociationIdsByAttributes: ")
+		}
+	}
+
+	// Map deleted UTXO AssociationIDs to keys.
+	deletedUtxoKeys := NewSet[string]([]string{})
+	for _, deletedUtxoAssociationId := range deletedUtxoAssociationIDs.ToSlice() {
+		deletedUtxoKey, err := _dbPostAssociationIdToKey(handle, snap, deletedUtxoAssociationId, prefixType)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "DBGetPostAssociationIdsByAttributes: ")
+		}
+		if deletedUtxoKey != nil {
+			deletedUtxoKeys.Add(string(deletedUtxoKey))
+		}
+	}
+
 	// Scan for all association IDs with the given key prefix.
-	_, valsFound := EnumerateKeysForPrefix(handle, keyPrefix)
-	var associationIDs []*BlockHash
+	_, valsFound, err := EnumerateKeysForPrefixWithLimitOffsetOrder(
+		handle,
+		keyPrefix,
+		associationQuery.Limit,
+		lastSeenKey,
+		associationQuery.SortDescending,
+		deletedUtxoKeys,
+	)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "DBGetPostAssociationsIdsByAttributes: ")
+	}
 
 	// Cast resulting values from bytes to association IDs.
+	associationIDs := NewSet[*BlockHash]([]*BlockHash{})
 	for _, valBytes := range valsFound {
 		associationID := &BlockHash{}
 		rr := bytes.NewReader(valBytes)
 		if exist, err := DecodeFromBytes(associationID, rr); !exist || err != nil {
-			return nil, errors.Wrapf(err, "DBGetPostAssociationIdsByAttributes: problem decoding association id: ")
+			return nil, nil, errors.Wrapf(err, "DBGetPostAssociationIdsByAttributes: problem decoding association id: ")
 		}
-		associationIDs = append(associationIDs, associationID)
+		associationIDs.Add(associationID)
+	}
+	return associationIDs, prefixType, nil
+}
+
+func _dbUserAssociationIdToKey(
+	handle *badger.DB,
+	snap *Snapshot,
+	associationID *BlockHash,
+	prefixType []byte,
+) ([]byte, error) {
+	// Converts a UserAssociationID to a db key.
+	associationEntry, err := DBGetUserAssociationByID(handle, snap, associationID)
+	if err != nil {
+		return nil, err
+	}
+	if associationEntry == nil {
+		return nil, nil
 	}
-	return associationIDs, nil
+	key, err := DBKeyForUserAssociationByPrefix(associationEntry, prefixType)
+	if err != nil {
+		return nil, err
+	}
+	return key, nil
```

Comment:
can we just return the response from the function?
```suggestion
	return DBKeyForUserAssociationByPrefix(associationEntry, prefixType)
```